back/engines/commercial/id_vienna_saml/app/lib/id_vienna_saml/citizen_saml_omniauth.rb

```diff
@@ -20,12 +20,16 @@ class CitizenSamlOmniauth
     # @return [Hash] The user attributes
     def profile_to_user_attrs(auth)
       attrs = auth.dig(:extra, :raw_info).to_h
+      email =  attrs.fetch('urn:oid:0.9.2342.19200300.100.1.3').first
+      locale = AppConfiguration.instance.settings.dig('core', 'locales').first
+      first_name = attrs.fetch('urn:oid:2.5.4.42', [generate_placeholder_from_email(email, :first)]).first
```

Comment:
This interested me. I think what this does is return the second argument for `fetch`, if the value of the first is `nil`.

Reading a little about this I came across a post that suggests this syntax will still invoke the evaluation of the second argument, even if it is never used, and that this form would avoid this:
```
pets.fetch(:dinosaur) { Dinosaur.raise_from_the_dead! }
```
Maybe this useful here?

Reply:
Oh good idea!